spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -52,37 +50,92 @@
 	 */
 	private final ConcurrentHashMap<String, Publisher> publishers = new ConcurrentHashMap<>();
 
-	private final ExecutorProvider executorProvider;
+	private ExecutorProvider executorProvider;
+
+	private TransportChannelProvider channelProvider;
+
+	private CredentialsProvider credentialsProvider;
 
-	private final TransportChannelProvider channelProvider;
+	private RetrySettings retrySettings;
 
-	private final CredentialsProvider credentialsProvider;
+	private BatchingSettings batchingSettings;
 
-	public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
-			ExecutorProvider executorProvider,
-			TransportChannelProvider channelProvider,
-			CredentialsProvider credentialsProvider) {
+	/**
+	 * @param projectIdProvider provides the GCP project ID
```

Comment:
FYI: just explain `@params` isn't not enough. There must a general description for methods.
When you list class method in JavaDocs page, that general description is present there in the table.
You see `@params` only when you jump into the method.

Anyway that is not Spring way to make end-user wrinkle from our not beautiful code 😄